dialogue-core/src/main/java/com/palantir/dialogue/core/NodeSelectionStrategyChannel.java

```diff
@@ -93,6 +95,11 @@ static LimitedChannel create(Config cf, ImmutableList<LimitedChannel> channels)
 
     @Override
     public Optional<ListenableFuture<Response>> maybeExecute(Endpoint endpoint, Request request) {
+        RoutingKey routingKey = request.attachments().getOrDefault(RoutingAttachments.ROUTING_KEY, null);
+        if (routingKey != null) {
+            return channels.get(routingKey.hostId().value()).maybeExecute(endpoint, request);
```

Comment:
On config update we may land on another node, but that's probably fine. We don't necessarily want to use an outdated config target either.

Reply:
Yeah, I am punting that. I think there needs to be an expectation that if a host stops being in the set of URLs, this sticky channel is just broken and there's not much we can do about it.

There's something to be said though about just indexes changing underneath somehow? Like if we received a config update that somehow shuffled the URLs? so probably want to switch this to a UUID as well, and have the node selecting channel just generate unique ones when it's created.

Reply:
Either way I'm going to punt since this should be a minor point, as long as we solve the routing and refreshing.